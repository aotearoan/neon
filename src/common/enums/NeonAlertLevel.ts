/**
 * Alert levels are the severity levels to be used with <a href="/feedback/alert#examples">NeonAlert & NeonToast</a>
 * components to inform the user of the severity of a message.
 */
export enum NeonAlertLevel {
  /** Information level, use for standard informational messages. */
  Info = 'info',
  /** Success level, use for indicating successful actions. */
  Success = 'success',
  /** Warning level, use for indicating a problem that the user <strong>can</strong> recover from. */
  Warn = 'warn',
  /** Error level, use for indicating a problem that the user <strong>cannot</strong> recover from. */
  Error = 'error',
}
