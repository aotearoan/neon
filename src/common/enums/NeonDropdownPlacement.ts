/**
 * Describes placement of dropdown related components like:
 * <ul class="no-style">
 *  <li><a href="/user-input/select">NeonSelect</a></li>
 *  <li><a href="/navigation/dropdown-menu">NeonDropdownMenu</a></li>
 *  <li><a href="/presentation/dropdown">NeonDropdown</a></li>
 * </ul>
 *
 * <p>The placement has a <strong>primary placement</strong> (the first direction) & a <strong>secondary placement</strong>
 * (the second direction). The primary placement is the location of the popup relative to the triggering element (e.g.
 * the select button). With the secondary placement defining the alignment relative to the trigger. E.g. BottomLeft will
 * open the dropdown below the trigger element with the left-hand edges aligned.</p>
 *
 * <p>The placement utilities are smart & dynamic. They listen to resize events & aim for best effort based on where the
 * dropdown is opened. E.g. in the case above with BottomLeft if there is not enough visible room to align the component
 * to the left then it will try to align to the right & likewise if there is not enough space below the trigger it will
 * align to the top (& reverse the option order in the case of a select).</p>
 */
export enum NeonDropdownPlacement {
  /** Placement above trigger element, aligned to the left */
  TopLeft = 'top-left',
  /** Placement above trigger element, aligned to the right */
  TopRight = 'top-right',
  /** Placement below trigger element, aligned to the left */
  BottomLeft = 'bottom-left',
  /** Placement below trigger element, aligned to the right */
  BottomRight = 'bottom-right',
  /** Placement to the left of the trigger element, aligned to the top */
  LeftTop = 'left-top',
  /** Placement to the left of the trigger element, aligned to the bottom */
  LeftBottom = 'left-bottom',
  /** Placement to the right of the trigger element, aligned to the top */
  RightTop = 'right-top',
  /** Placement to the right of the trigger element, aligned to the bottom */
  RightBottom = 'right-bottom',
}
