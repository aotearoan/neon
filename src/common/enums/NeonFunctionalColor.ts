/**
 * The core color palettes available for styling components in Neon. See the design section
 * <a href="/design/color">Color</a> page for more details on color palettes.
 */
export enum NeonFunctionalColor {
  /** Low contrast is a more subtle neutral palette which can be used for secondary actions & supporting text. */
  LowContrast = 'low-contrast',
  /** Neutral is the base greyscale palette. */
  Neutral = 'neutral',
  /**
   * High contrast is a higher contrast palette than can be used for increased accessibility or for a stronger black &
   * white application.
   * */
  HighContrast = 'high-contrast',
  /** Brand is the main brand color palette, this is useful for branded CTAs, text & controls. */
  Brand = 'brand',
  /** Primary is the main application color palette. For most applications this will be used by default. */
  Primary = 'primary',
  /** Info can be used to convey informational context, e.g. notes & alerts */
  Info = 'info',
  /** Success can be used to convey successful actions. */
  Success = 'success',
  /** Warn can be used to convey issues/actions that are less severe but may still impact the user. */
  Warn = 'warn',
  /** Error can be used to convey issues/actions that are severe & that the user may not be able to recover from. */
  Error = 'error',
}
