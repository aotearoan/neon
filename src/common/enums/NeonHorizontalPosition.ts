/**
 * Describes the horizontal positioning of certain elements in the context of a component. An example would be an icon
 * in a button or label as well as the label for a checkbox or switch.
 */
export enum NeonHorizontalPosition {
  /** Left horizontal positioning. */
  Left = 'left',
  /** Right horizontal positioning. */
  Right = 'right',
}
