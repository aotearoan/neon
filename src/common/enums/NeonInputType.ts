/**
 * Corresponds to the <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types">HTML input
 * types</a> for use with <a href="/user-input/input">NeonInput</a>.
 *
 * NOTE: prefer the specific components for Number, Password, Date, File, Search fields.
 */
export enum NeonInputType {
  /** Date input for selecting dates from a calendar. Use <a href="/user-input/date-picker">NeonDatePicker</a>. */
  Date = 'date',
  /** Date time input for selecting dates including a specific time. */
  DateTime = 'datetime-local',
  /** Email input for entering email addresses. */
  Email = 'email',
  /**
   * File input for opening a file dialog. Use <a href="/user-input/file">NeonDatePicker</a> or
   * <a href="/user-input/drop-zone">NeonDropZone</a>.
   */
  File = 'file',
  /** Number input for entering numeric values. Use <a href="/user-input/number">NeonNumber</a> */
  Number = 'number',
  /** Password input for entering passwords. Use <a href="/user-input/password">NeonPassword</a> */
  Password = 'password',
  /**
   * Range input for a slider control. Use <a href="/user-input/slider">NeonSlider</a> or
   * <a href="/user-input/range-slider">NeonRangeSlider</a>
   */
  Range = 'range',
  /** Search input for searching. Use <a href="/user-input/search">NeonSearch</a> */
  Search = 'search',
  /** Telephone number input. */
  Tel = 'tel',
  /** Text input for single or multiline text input. */
  Text = 'text',
  /** Time input. */
  Time = 'time',
  /** URL input for entering web addresses. */
  Url = 'url',
}
