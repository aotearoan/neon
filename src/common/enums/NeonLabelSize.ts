/**
 * The size options for the <a href="/presentation/label">NeonLabel</a> component.
 */
export enum NeonLabelSize {
  /** Extra extra small, default is 16 rem. */
  ExtraExtraSmall = 'xxs',
  /** Extra small, default is 20 rem. */
  ExtraSmall = 'xs',
  /** Small, default is 28 rem. */
  Small = 's',
  /** Medium, default is 36 rem. */
  Medium = 'm',
  /** Large, default is 44 rem. */
  Large = 'l',
}
