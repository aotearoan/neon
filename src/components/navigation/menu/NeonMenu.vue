<template>
  <nav ref="menuWrapper" class="neon-menu__wrapper">
    <ul :class="[`neon-menu--${color}`, `neon-menu--${size}`]" class="neon-menu no-style" role="menubar">
      <li
        v-for="item in menu"
        :key="item.key"
        ref="menuItem"
        :class="{ 'neon-menu__item--disabled': item.disabled }"
        class="neon-menu__item"
      >
        <neon-dropdown-menu
          v-if="item.children"
          :class="{ 'router-link-active': routeMatches(item.href) }"
          :color="color"
          :disabled="item.disabled"
          :icon="item.icon"
          :label="item.label"
          :model="item.children"
          :openOnHover="!item.disabled"
          :size="size"
          dropdown-style="text-button"
        />
        <neon-link
          v-else
          :href="item.href"
          :no-style="true"
          :tabindex="item.disabled ? -1 : 0"
          outline-style="none"
          role="menuitem"
          @click="!item.disabled && onClick(item.key)"
          @keydown.enter="!item.disabled && onClick(item.key)"
          @keydown.space="!item.disabled && onClick(item.key)"
          @keypress.space.prevent=""
        >
          <neon-icon v-if="item.icon" :name="item.icon" class="neon-menu__item-icon" color="neutral" />
          <span class="neon-menu__item-label">{{ item.label }}</span>
        </neon-link>
      </li>
    </ul>
    <neon-dropdown-menu
      :class="{ 'neon-menu__responsive-menu--hidden': responsiveMenuItems.length === 0 }"
      :color="color"
      :indicator="false"
      :model="responsiveMenuItems"
      :openOnHover="true"
      :size="size"
      class="neon-menu__responsive-menu"
      dropdown-style="text-button"
      icon="ellipsis"
      placement="bottom-right"
      @button-ref="responsiveButton = $event"
    />
  </nav>
</template>

<script lang="ts" src="./NeonMenu.ts" />
