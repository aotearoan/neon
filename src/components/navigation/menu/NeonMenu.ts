import { defineComponent, nextTick, onMounted, onUnmounted, ref } from 'vue';
import type { NeonMenuModel } from '@/common/models/NeonMenuModel';
import NeonLink from '@/components/navigation/link/NeonLink.vue';
import NeonDropdownMenu from '@/components/navigation/dropdown-menu/NeonDropdownMenu.vue';
import { NeonFunctionalColor } from '@/common/enums/NeonFunctionalColor';
import { NeonSize } from '@/common/enums/NeonSize';
import type { NeonPriorityMenuItem } from './NeonPriorityMenuItem';
import NeonIcon from '@/components/presentation/icon/NeonIcon.vue';
import { useRoute } from 'vue-router';

/**
 * <p>
 *   <strong>NeonMenu</strong> is a responsive aware menu that progressively collapses options into the mobile menu as
 *   the screen size gets smaller. This is a more flexible option than moving directly to a hamburger menu at tablet and
 *   below. It is a best effort approach to keep displaying the most important menu items for as long as possible,
 *   giving the user a better experience.
 * </p>
 */
export default defineComponent({
  name: 'NeonMenu',
  components: {
    NeonDropdownMenu,
    NeonLink,
    NeonIcon,
  },
  props: {
    /**
     * The menu configuration. This can have two levels, i.e. a top level horizontal menu and, if required, a dropdown
     * menu containing the second level. The highlighted 'active' menu is determined by the current Vue route.
     */
    menu: { type: Array as () => Array<NeonMenuModel>, required: true },
    /**
     * The menu highlight color (excludes low-contrast and neutral).
     */
    color: { type: String as () => NeonFunctionalColor, default: NeonFunctionalColor.Brand },
    /**
     * The menu size.
     */
    size: { type: String as () => NeonSize, default: NeonSize.Large },
    /**
     * Whether to enable the priority menu which automatically calculates the available screen space and displays
     * as many of the menu items as possible, moving the remaining items into the overflow menu.
     */
    priorityMenuEnabled: { type: Boolean, default: true },
  },
  emits: [
    /**
     * Emitted when a user clicks on a menu item.
     * @type {string} the key of the menu item clicked.
     */
    'click',
  ],
  setup(props, { emit }) {
    const route = useRoute();
    const menuWrapper = ref<HTMLElement | null>(null);
    const menuItem = ref<Array<HTMLElement> | null>(null);
    const responsiveButton = ref<HTMLElement | null>(null);

    const menuItems = ref<Array<NeonPriorityMenuItem>>([]);
    const responsiveMenuItems = ref<Array<NeonMenuModel>>([]);
    const visible = ref<Array<string>>([]);

    const getWidth = (el: HTMLElement) => {
      const styles = window.getComputedStyle(el);
      const margin = parseFloat(styles.marginLeft || '0') + parseFloat(styles.marginRight || '0');
      return Math.ceil(el.offsetWidth + margin);
    };

    const toPriorityMenuItem = (key: string, element: HTMLElement): NeonPriorityMenuItem => {
      return {
        key,
        element,
        width: getWidth(element),
      };
    };

    const initMenuItems = () => {
      const results = props.menu.map(
        (item, index) => (menuItem.value && toPriorityMenuItem(item.key, menuItem.value[index])) || null,
      );
      menuItems.value = results.filter((item) => item !== null) as NeonPriorityMenuItem[];
    };

    const routeMatches = (path?: string) => {
      return path && route?.path.indexOf(path) >= 0;
    };

    const onClick = (key: string) => {
      emit('click', key);
    };

    const determineVisibleMenuItems = (
      menuWidth: number,
      responsiveMenuWidth: number,
      menuItems: NeonPriorityMenuItem[],
    ) => {
      const itemWidthSum = menuItems.map((item) => item.width).reduce((acc: number, width) => (acc ? acc + width : 0));

      // no responsive menu
      if (itemWidthSum <= menuWidth - responsiveMenuWidth) {
        return menuItems.map((menuItem) => menuItem.key);
      }

      // with responsive menu
      let availableWidth = menuWidth - responsiveMenuWidth;
      let visible: string[] = [];

      for (let i = 0; i < menuItems.length; i = i + 1) {
        const menuItem = menuItems[i];
        if (availableWidth < menuItem.width) {
          // if the second menu item should be hidden, also hide the first one to improve how it looks
          if (i === 1) {
            visible = visible.filter((key) => key !== menuItems[0].key);
          }
          break;
        }

        availableWidth = availableWidth - menuItem.width;
        visible.push(menuItem.key);
      }

      return visible;
    };

    const refreshVisibleMenu = async () => {
      await nextTick();
      const menuWidth = (menuWrapper.value && getWidth(menuWrapper.value)) || 0;
      const responsiveMenuWidth = (responsiveButton.value && getWidth(responsiveButton.value)) || 0;
      visible.value = determineVisibleMenuItems(menuWidth, responsiveMenuWidth, menuItems.value);

      responsiveMenuItems.value = props.menu
        .filter((item) => visible.value.indexOf(item.key) < 0)
        .flatMap((item) => [
          { ...item, isGroup: item.children && item.children.length > 0 },
          ...(item.children || []).map((item) => ({ ...item, grouped: true })),
        ]);

      menuItems.value.forEach((item) => {
        item.element.hidden = visible.value.indexOf(item.key) < 0;
      });

      if (responsiveButton.value) {
        responsiveButton.value.hidden = menuItems.value.length === visible.value.length;
      }
    };

    onMounted(async () => {
      if (props.priorityMenuEnabled) {
        await nextTick();
        initMenuItems();
        await refreshVisibleMenu();
        window.addEventListener('resize', refreshVisibleMenu);
      }
    });

    onUnmounted(() => {
      if (props.priorityMenuEnabled) {
        window.removeEventListener('resize', refreshVisibleMenu);
      }
    });

    return {
      menuWrapper,
      menuItem,
      menuItems,
      responsiveButton,
      responsiveMenuItems,
      visible,
      onClick,
      routeMatches,
    };
  },
});
