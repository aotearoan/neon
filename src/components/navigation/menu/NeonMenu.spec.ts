import type { RenderResult } from '@testing-library/vue';
import { render } from '@testing-library/vue';
import NeonMenu from '../menu/NeonMenu.vue';
import { NeonFunctionalColor } from '../../../common/enums/NeonFunctionalColor';
import { NeonSize } from '../../../common/enums/NeonSize';

describe('NeonMenu', () => {
  const menu = [
    {
      key: 'action-menu',
      label: 'Action Menu',
      href: '/navigation/action-menu',
    },
    {
      key: 'link',
      label: 'Link',
      href: '/navigation/link',
    },
    {
      key: 'menu',
      label: 'Menu',
      href: '/navigation/menu',
    },
    {
      key: 'click-link',
      icon: 'contrast',
      label: 'Click link',
    },
    {
      key: 'disabled',
      label: 'Disabled',
      disabled: true,
    },
    {
      key: 'tree-menu',
      label: 'Tree Menu',
      href: '/navigation/tree-menu',
      children: [
        {
          key: 'tree-menu-description',
          label: 'Description',
          href: '/navigation/tree-menu#description',
        },
        {
          key: 'tree-menu-api',
          label: 'API',
          href: '/navigation/tree-menu#api',
        },
        {
          key: 'tree-menu-examples',
          label: 'Examples',
          href: '/navigation/tree-menu#examples',
        },
      ],
    },
  ];

  let harness: RenderResult;

  beforeEach(() => {
    harness = render(NeonMenu, {
      props: {
        menu,
      },
      global: {
        stubs: ['router-link'],
        mocks: {
          route: { path: '.' },
        },
      },
    });
  });

  it('renders menu', () => {
    // given
    const { html } = harness;
    // when / then
    expect(html()).toMatchSnapshot();
  });

  it('renders default color', () => {
    // given
    const { container } = harness;
    // when / then
    expect(container.querySelector('.neon-menu--brand')).toBeDefined();
    expect(container.querySelectorAll('.neon-dropdown-menu--brand').length).toEqual(2);
  });

  it('renders color', async () => {
    // given
    const { container, rerender } = harness;
    await rerender({ color: NeonFunctionalColor.Primary });
    // when / then
    expect(container.querySelector('.neon-menu--primary')).toBeDefined();
    expect(container.querySelectorAll('.neon-dropdown-menu--primary').length).toEqual(2);
  });

  it('renders default size', () => {
    // given
    const { container } = harness;
    // when / then
    expect(container.querySelector('.neon-menu--l')).toBeDefined();
    expect(container.querySelectorAll('.neon-dropdown--l').length).toEqual(2);
  });

  it('renders size', async () => {
    // given
    const { container, rerender } = harness;
    await rerender({ size: NeonSize.Small });
    // when / then
    expect(container.querySelector('.neon-menu--s')).toBeDefined();
    expect(container.querySelectorAll('.neon-dropdown--s').length).toEqual(2);
  });

  it('removes resize handler on destroy', () => {
    // given
    window.addEventListener = jest.fn();
    window.removeEventListener = jest.fn();
    const { unmount } = harness;
    // when
    unmount();
    // then
    expect(window.addEventListener).toHaveBeenCalledTimes(5);
    expect(window.removeEventListener).toHaveBeenCalledTimes(5);
  });

  it('does not remove resize handler on destroy', async () => {
    // given
    window.addEventListener = jest.fn();
    window.removeEventListener = jest.fn();
    const { unmount, rerender } = harness;
    await rerender({ priorityMenuEnabled: false });
    unmount();
    // then
    expect(window.addEventListener).toHaveBeenCalledTimes(0);
    expect(window.removeEventListener).toHaveBeenCalledTimes(0);
  });

  it('emits click event', () => {
    // given
    const { container, emitted } = harness;
    // when
    const item = container.querySelector('.neon-menu__item:nth-child(4) .neon-link') as HTMLElement;
    item?.click();
    // then
    expect(emitted().click[0]).toEqual([menu[3].key]);
  });
});
