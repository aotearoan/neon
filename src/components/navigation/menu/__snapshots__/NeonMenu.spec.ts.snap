// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonMenu renders menu 1`] = `
<nav class="neon-menu__wrapper">
  <ul class="neon-menu--brand neon-menu--l neon-menu no-style" role="menubar">
    <li class="neon-menu__item">
      <!--v-if--><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-menu__item-label">Action Menu</span></span></a>
    </li>
    <li class="neon-menu__item">
      <!--v-if--><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-menu__item-label">Link</span></span></a>
    </li>
    <li class="neon-menu__item">
      <!--v-if--><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-menu__item-label">Menu</span></span></a>
    </li>
    <li class="neon-menu__item">
      <!--v-if--><a class="neon-link--no-style neon-link--outline-none neon-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><div class="neon-icon--neutral neon-icon neon-menu__item-icon neon-menu__item-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 512 512">
    <circle class="neon-svg--stroke" cx="256" cy="256" r="208" style="fill:none;stroke-linejoin:round;stroke-width:32px"></circle>
    <path class="neon-svg--fill" d="M256,464C141.12,464,48,370.88,48,256S141.12,48,256,48Z"></path>
</svg>
</div><span class="neon-menu__item-label">Click link</span></span></a>
    </li>
    <li class="neon-menu__item--disabled neon-menu__item">
      <!--v-if--><a class="neon-link--no-style neon-link--outline-none neon-link" tabindex="-1" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-menu__item-label">Disabled</span></span></a>
    </li>
    <li class="neon-menu__item">
      <div class="neon-dropdown--with-indicator neon-dropdown--open-on-hover neon-dropdown--l neon-dropdown--brand neon-dropdown neon-dropdown-menu--brand neon-dropdown-menu">
        <div>
          <!-- @slot optionally replace the dropdown button with a custom control -->
          <div class="neon-dropdown__button-wrapper"><button class="neon-button--l neon-button--brand neon-button--text neon-button--state-ready neon-button neon-dropdown__button neon-dropdown__button" aria-expanded="false" aria-haspopup="true">
              <!--v-if--><span class="neon-button__label">Tree Menu</span>
              <div class="neon-expansion-indicator neon-expansion-indicator--brand neon-button__indicator" role="presentation"><svg width="16" height="16" viewbox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="xMidYMid">
                  <g class="neon-expansion-indicator__container">
                    <line class="neon-svg--stroke neon-expansion-indicator__line1" x1="2" y1="8" x2="8" y2="8"></line>
                    <line class="neon-svg--stroke neon-expansion-indicator__line2" x1="14" y1="8" x2="8" y2="8"></line>
                  </g>
                </svg></div>
            </button></div>
        </div>
        <div class="neon-dropdown__click-blocker"></div>
        <div class="neon-dropdown__content--bottom-left neon-dropdown__content">
          <!-- @slot The content of the open dropdown -->
          <ul class="no-style neon-dropdown-menu__items" role="menu">
            <li class="neon-dropdown-menu__item--l neon-dropdown-menu__item"><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-dropdown-menu__item-label">Description</span></span></a></li>
            <li class="neon-dropdown-menu__item--l neon-dropdown-menu__item"><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-dropdown-menu__item-label">API</span></span></a></li>
            <li class="neon-dropdown-menu__item--l neon-dropdown-menu__item"><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-dropdown-menu__item-label">Examples</span></span></a></li>
          </ul>
        </div>
      </div><a href="/test" class="neon-link--no-style neon-link--active neon-link--outline-none neon-link neon-link--router-link" tabindex="0" role="menuitem"><span class="neon-link__label"><!-- @slot the content of the link --><!--v-if--><span class="neon-menu__item-label">Tree Menu</span></span></a>
    </li>
  </ul>
  <div class="neon-dropdown--open-on-hover neon-dropdown--icon-only neon-dropdown--l neon-dropdown--brand neon-dropdown neon-dropdown-menu--brand neon-dropdown-menu neon-menu__responsive-menu--hidden neon-menu__responsive-menu neon-menu__responsive-menu--hidden neon-menu__responsive-menu">
    <div>
      <!-- @slot optionally replace the dropdown button with a custom control -->
      <div class="neon-dropdown__button-wrapper"><button class="neon-button--l neon-button--brand neon-button--text neon-button--state-ready neon-button--with-icon neon-button--icon-only neon-button neon-dropdown__button neon-dropdown__button" aria-expanded="false" aria-haspopup="true">
          <div class="neon-icon--brand neon-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="100%" height="100%">
              <path class="neon-svg--fill" d="M8 9a1.5 1.5 0 100-3 1.5 1.5 0 000 3zM1.5 9a1.5 1.5 0 100-3 1.5 1.5 0 000 3zm13 0a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"></path>
            </svg>
          </div>
          <!--v-if-->
          <!--v-if-->
        </button></div>
    </div>
    <div class="neon-dropdown__click-blocker"></div>
    <div class="neon-dropdown__content--bottom-right neon-dropdown__content">
      <!-- @slot The content of the open dropdown -->
      <ul class="no-style neon-dropdown-menu__items" role="menu"></ul>
    </div>
  </div>
</nav>
`;
